### DISCUSSION TITLE
What is the dasel equivalent of this python-yq command?

### DISCUSSION THREAD

**Discussion by sdemura** [TomWright/dasel#234] [Q&A] [answered]
The following `jq` command will print out the values of all keys with the name 'image' and type of string. The file is a multi-doc yaml of kubernetes definitions -- it's an assortment of jobs, deployment, service, crd and crd definitions.

``` 
yq -r '.. | ."image"? | select(. != null and type=="string")' test.yaml
```

Based on the documentation, I would suspect to be able to do something like this, but it doesn't work:

```
 ❯ dasel select -f test.yaml -m '.(key=image)'
Error: could not query multiple node: could not find multiple value: no value found for selector: .(-=image): 
```

**Comment by sdemura** [comment]
I think, that the real issue here is that if dasel does not find a specific key it errors out. I think that's okay in a single document sense, but for multiple documents, there's no guarantee they follow the same schema.

In other words, does, or can dasel support the `?` (optional, if exists) directive like JQ? does?

**Reply by TomWright** [reply]
It doesn't support optional selectors right now but it's definitely a worthwhile feature I want to add

**Comment by TomWright** [comment, marked as answer]
This feature is now available in `v1.26.0`.
